docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+```
+
+The controller will automatically add the versioning-related environment variables during migration.
+
+## Understanding the Differences
+
+### Before: Unversioned Workers
+
+```yaml
+# Single deployment, all workers run the same code
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.3
+        env:
+        - name: TEMPORAL_HOST_PORT
+          value: "production.tmprl.cloud:7233"
+        - name: TEMPORAL_NAMESPACE
+          value: "production"
+        # No versioning environment variables
+```
+
+**Current Deployment Process:**
+1. Build new worker image
+2. Update Deployment with new image
+3. Kubernetes rolls out new pods, terminating old ones
+4. **Risk**: Running workflows may fail if code changes break compatibility
+
+### After: Versioned Workers with Controller
+
+```yaml
+# Single CRD manages multiple versions automatically
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  cutover:
+    strategy: Progressive  # Gradual rollout of new versions
+    steps:
+      - rampPercentage: 10
+        pauseDuration: 5m
+      - rampPercentage: 50
+        pauseDuration: 10m
+  sunset:
+    scaledownDelay: 1h
+    deleteDelay: 24h
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.4  # Update this to deploy new versions
+        # Note: Controller automatically adds versioning environment variables:
+        # TEMPORAL_HOST_PORT, TEMPORAL_NAMESPACE, TEMPORAL_DEPLOYMENT_NAME, WORKER_BUILD_ID
+```
+
+**New Deployment Process:**
+1. Build new worker image  
+2. Update `TemporalWorkerDeployment` CRD with new image
+3. Controller creates new Kubernetes `Deployment` for the new version
+4. Controller gradually routes new workflows to new version
+5. Old version continues handling existing workflows until they complete
+6. **Safety**: No disruption to running workflows, automated rollout control
+
+**Key Benefits:**
+- ✅ **Zero-disruption deployments** - Running workflows continue on original version
+- ✅ **Automated version management** - Controller handles registration and routing
+- ✅ **Progressive rollouts** - Gradual traffic shifting with automatic pause points
+- ✅ **Easy rollbacks** - Instantly route new workflows back to previous version
+- ✅ **Workflow continuity** - Deterministic execution preserved across deployments
+
+## Migration Strategy
+
+### Safe Migration Approach
+
+The migration from unversioned to versioned workflows requires careful planning to avoid disrupting running workflows. The key is to transition gradually while maintaining workflow continuity.
+
+**Key Principles:**
+- **Start with Manual strategy** - Prevents automatic promotions during initial setup
```

Comment:
I think we want folks to try out the thing that they will want to do long-term. So ideally they should try out Progressive, but perhaps with a more conservative rollout policy than they might do later on.

If people start with Manual, they'll have to do "wasted" work developing a system to call the upgrade APIs with the CLI or SDK, which is work that won't be re-used in production, and prone to errors.

Idk that's just my take. I think in an ephemeral environment, folks can happily play with Manual mode all they want, but my recomendation for migrating brownfield environment would be to /add/ a TWD while still deploying the unversioned pollers.

Reply:
just bumping this, I think the "start with manual strategy" recommendation should be removed! but happy to hear your reasoning if you disagree

Reply:
Adjusted to progressive.